images/linux/scripts/base/apt-mock.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+prefix=/usr/local/bin
+
+for tool in apt apt-get apt-fast;do
+  real_tool=`which $tool`
+  cat >$prefix/$tool <<EOT
+#!/bin/sh
+
+i=1
+while [ \$i -le 10 ];do
+  $real_tool "\$@"
```

Comment:
Looks like this logic will also retry commands like `apt install tool1`. Do we really want to do it?
I think we should be more picky for apt retrying and don't retry any command blindly

Reply:
it is intended to retry `apt install tool1` in fact. Any apt related command attempts to lock the database and fails if the updater already locked it. 

There's nothing serious affected by this retry, so there nothing to care about.

Reply:
I disagree.
Actually, impact can be huge.
You leave these files on machine after build. It means that it will be applied to all customers builds on image.
It means that all customers' builds will try to rerun any apt command 10 times instead of immediate failure that will affect customers' build duration and build cost.

Reply:
When customer uses image, they expect `apt` to work as `apt` on their local machine. These changes override their apt tool and replace it with our retry-able implementation. I don't think that it is correct behavior.

Reply:
I still don't understand why we apply this fix to absolutely all apt commands if actually only `apt clean` fails in the single place.
As I understand, @Darleev is going to fix any other random apt failures in https://github.com/actions/virtual-environments/pull/1761 so I am trying to understand if you are fixing the same issue or different one.

Reply:
yes, it was intentionally developed to be a workaround till this issue https://github.com/Azure/azure-linux-extensions/issues/1238 resolved. 

The problem with apt is not limited to clean task, it happens sporadically during any apt use that interferes with omsagent update.